src/libs/relay/conduit_relay_io_csv.cpp

```diff
@@ -0,0 +1,663 @@
+// Copyright (c) Lawrence Livermore National Security, LLC and other Conduit
+// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
+// other details. No copyright assignment is required to contribute to Conduit.
+
+//-----------------------------------------------------------------------------
+///
+/// file: conduit_relay_io_csv.cpp
+///
+//-----------------------------------------------------------------------------
+#include "conduit_relay_io_csv.hpp"
+
+#include <algorithm>
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <utility>
+#include <type_traits>
+
+#include "conduit_log.hpp"
+#include "conduit_blueprint_table.hpp"
+
+using conduit::utils::log::quote;
+
+// Some constants used by writer/reader
+
+// Child separator used by writer to name columns
+const char child_sep = '/';
+
+// Default argument to trim
+const char *whitespace = " \t\n\r\f\v";
+
+// Prefix used for file names when the given table collection is a list
+const std::string table_list_prefix = "table_list_";
+
+//-----------------------------------------------------------------------------
+// -- begin conduit:: --
+//-----------------------------------------------------------------------------
+namespace conduit
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::relay --
+//-----------------------------------------------------------------------------
+namespace relay
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::relay::io --
+//-----------------------------------------------------------------------------
+namespace io
+{
+
+// Static functions, internal types
+
+//-----------------------------------------------------------------------------
+static index_t
+get_nrows(const Node &table)
+{
+    const Node &values = table["values"];
+    index_t retval = 0;
+    if(values.number_of_children() > 0)
+    {
+        const Node &ref = values[0];
+        const index_t nc = ref.number_of_children();
+        retval = nc > 0
+            ? ref[0].dtype().number_of_elements()
+            : ref.dtype().number_of_elements();
+    }
+    return retval;
```

Comment:
should this be a property method of the table blueprint?

Reply:
Yea this would make sense.

I was probably hesitant to do this because the table blueprint supports multiple tables in one node (like a multi domain mesh), and this would be something that only works on a single table. The mesh blueprint supports a ton of operations like that so I don't see why a table shouldn't.